src/mapml/layers/MapLayer.js

```diff
@@ -861,31 +861,62 @@ export var MapMLLayer = L.Layer.extend({
                 M.parseStylesheetAsHTML(mapml, base, layer._container);
                 var styleLinks = mapml.querySelectorAll('map-link[rel=style],map-link[rel="self style"],map-link[rel="style self"]');
                 if (styleLinks.length > 1) {
+                  var changeStyle = function (e) {
+                    var href = getStyleURL(e);
+                    if (e.type === 'focusout' && !isStyleOption(e.relatedTarget) && href != layer._href) {
+                      layer.fire('changestyle', {src: href}, false);
+                    }
+                    function isStyleOption(item) {
+                      var form = e.target.form,  
+                          stylesRadioNodeList = form.elements["styles-"+L.stamp(layer)],
+                          isRB = false;
+                      for(var i=0;i<stylesRadioNodeList.length;i++) {
+                        if (item === stylesRadioNodeList[i]) {
+                          isRB = true;
+                          break;
+                        }
+                      }
+                      return isRB;
+                    }
+                    function getStyleURL(e) {
+                      var form = e.target.form,
+                      stylesRadioNodeList = form.elements["styles-"+L.stamp(layer)],
+                      href;
+                      for (var i=0;i < stylesRadioNodeList.length;i++) {
+                        if (stylesRadioNodeList[i].checked) {
+                          href =  stylesRadioNodeList[i].getAttribute('data-href');
+                          break;
+                        }
+                      }
+                      return href;
+                    }
+                  };
                   var stylesControl = document.createElement('details'),
-                  stylesControlSummary = document.createElement('summary');
+                  stylesControlSummary = document.createElement('summary'),
+                  stylesForm = document.createElement('form');
```

Comment:
Is an additional (form) element required?

Reply:
Would have to modify the following selector in this case, as it causes a small misalignment between the label and input:

https://github.com/Maps4HTML/Web-Map-Custom-Element/blob/816f7c19372bae1da571f74656b38a555d08bbab/src/mapml.css#L226-L231